### PR TITLE
Check exclusive

### LINKED ISSUE #374: Use `rlang::check_exclusive()`

### PR DESCRIPTION
**Pull request by jonthegeek**
Closes #374 

I updated all other errors in `make_selector()` to what I believe to be the current preferred style, but did not update everything in `find_href()` in case I'm wrong about the preference.

### CODE REVIEW

File: R/selectors.R [hunk 1]
```diff
@@ -78,30 +78,32 @@ html_elements <- function(x, css, xpath) {
...
+  xml2::xml_find_first(x, xpath)
 }
 
-make_selector <- function(css, xpath) {
-  if (missing(css) && missing(xpath))
-    stop("Please supply one of css or xpath", call. = FALSE)
-  if (!missing(css) && !missing(xpath))
-    stop("Please supply css or xpath, not both", call. = FALSE)
+make_selector <- function(css, xpath, call = rlang::caller_env()) {
+  switch(
```
Comment (hadley) [inline R/selectors.R:92]: I'd suggest keeping the basic flow of the function the same, just calling `check_exclusive()` (without the `rlang::`) first. I'd suggest you leave the string checking messages as is in this PR, as those would be better handled with the standalone rlang input checkers.

File: R/selectors.R [hunk 2]
```diff
@@ -78,30 +78,32 @@ html_elements <- function(x, css, xpath) {
 
 #' @export
 html_elements.default <- function(x, css, xpath) {
-  xml2::xml_find_all(x, make_selector(css, xpath))
+  xpath <- make_selector(css, xpath)
```
Comment (hadley) [inline R/selectors.R:81]: I think these functions can stay the same too.

**Review by hadley — APPROVED** [review]
Thanks for working on this! Let me know if you're interested in finishing this off, otherwise I'm happy to do so.

**Comment by jonthegeek** [comment]
I can finish it off in a day or two! 

I'll need to remember why I made those changes you called out to see if I want to argue for them; right now I can't remember, but I assume I had an argument for them at the time.

I'll likely have more questions about what you want when I get a chance to dig into this!

**Comment by jonthegeek** [comment]
Should I update the workflow to match something more recently updated (eg [httr2](https://github.com/r-lib/httr2/blob/main/.github/workflows/R-CMD-check.yaml)) while I'm at it, to kill those `ubuntu-18.04` failures?

**Comment by hadley** [comment]
I don't see the failures on main, so you should just be able to rebase/merge from main.

**Comment by hadley** [comment]
Actually, I just started converting the error to use cli, so it might be just as easy for me to do `check_exclusive()` while I'm in here.

**Comment by jonthegeek** [comment]
Ok, I won't be offended if you go ahead and clear this out without me! I'll be happy to review, and I'll still try to figure out why I felt the need to change other things!

**Comment by hadley** [comment]
Done in #386. I'm sure there's still plenty of room for improvement though.

**Pull request closed without merging**